doc/rm/c_cpp_coding_style.md

```diff
@@ -139,6 +139,31 @@ Example:
 int create_rainbow(int pots_of_gold, int unicorns);
 ```
 
+### Polyglot headers
+
+***Headers intended to be included from both languages must contain `extern` guards; `#include`s should not be wrapped in `extern "C" {}`.***
+
+A *polyglot header* is a header file that can be safely included in either a `.c` or `.cc` file.
+In particular, this means that the file must not depend on any of the places where C and C++ semantics disagree.
+
+Such files must be explictly marked with `extern` guards like the following, starting after the file's `#include`s.
+```
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Declarations...
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif
+```
+Moreover, all `#includes` must also be of other polyglot headers.
```

Comment:
Note to match our discussion: system headers are not required to be polyglot.

Reply:
Actually, I was thinking about this... at a minimum, we probably want `stdbool.h` to become an empty file `#ifdef __cplusplus`. I don't think that affects what you said, but does that seem right to you? Right now I think that including `stdbool.h` silently breaks some implicit conversions around `bool` in C++.